libpod/runtime_ctr.go

```diff
@@ -54,19 +54,60 @@ func (r *Runtime) NewContainer(ctx context.Context, rSpec *spec.Spec, options ..
 		}
 	}
 
-	ctr.valid = true
-	ctr.state.State = ContainerStateConfigured
-	ctr.runtime = r
-
 	var pod *Pod
+	// Note: I added option handling here rather in pkg/spec/createconfig to
```

Comment:
The right place is with the container, but not here - do this as part of `generateSpec()` in container_internal_linux.go